core/src/impl/Kokkos_SharedAlloc.hpp

```diff
@@ -252,14 +252,14 @@ class SharedAllocationRecordCommon : public SharedAllocationRecord<void, void> {
   ~SharedAllocationRecordCommon();
   template <class ExecutionSpace>
   SharedAllocationRecordCommon(
-      ExecutionSpace const&, MemorySpace const& space, std::string const& label,
-      std::size_t alloc_size,
+      ExecutionSpace const& exec, MemorySpace const& space,
+      std::string const& label, std::size_t alloc_size,
```

Comment:
IMO using `exec_space` and `mem_space` since both are "spaces" would be better. Or `exec` and `mem`. But maybe this is more common in the code than I am aware that `space` implies memory space.

Reply:
`SharedAllocationRecord` is probably the only place where memory space instances are used. If there are any other changes requested, I'm happy to change. Otherwise, I'd prefer to possibly improve in a different pul request so we can make progress with pull requests that depend on this one.